sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -865,29 +993,61 @@ static void *_lwmac_thread(void *args)
                 msg_reply(&msg, &reply);
                 break;
```

Comment:
This all can go into `gnrc_netif2_ops_t::set()` so you don't need the thread anymore (and if you call `gnrc_netif2_set_from_netdev()` in default you don't have to add any of the `/////// netif2` stuff added above).